azurerm/resource_arm_iothub_resource_provider.go

```diff
@@ -0,0 +1,261 @@
+package azurerm
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/iothub/mgmt/2017-07-01/devices"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceArmIotHub() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmIotHubCreateAndUpdate,
+		Read:   resourceArmIotHubRead,
+		Update: resourceArmIotHubCreateAndUpdate,
+		Delete: resourceArmIotHubDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"tags":     tagsSchema(),
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+			"sku": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(devices.F1),
+								string(devices.S1),
+								string(devices.S2),
+								string(devices.S3),
+							}, true),
+						},
+
+						"tier": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(devices.Free),
+								string(devices.Standard),
+							}, true),
+						},
+
+						"capacity": {
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntAtLeast(1),
+						},
+					},
+				},
+			},
+
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"etag": {
```

Comment:
I think it might be better not to expose this field (since, it's handled internally within Azure?) and could cause user confusion - could we remove it?